lib/usdUfe/ufe/UsdUndoSetSceneItemMetadataCommand.cpp

```diff
@@ -80,7 +84,8 @@ void SetSceneItemMetadataCommand::setGroupMetadata()
         PXR_NS::TfToken        fullKey(prefixlessGroupName + std::string(":") + _key);
```

Comment:
@pierrebai-adsk, @AramAzhari-adsk,
Should I enable edit routing here? "SessionLayer-" customData is always expected to be authored in the session, or is this a default behavior that could be overriden ? 
I noticed this is used for `Autodesk:Text Color`, it would be nice if we could author it in an identified layer when changed from AE.

Reply:
One option would be to call getLayer() on the edit router context and only set it to the session layer if it returns nullptr. OTOH, for composite operation or if sessions metadata is set in the middle of another operation (say, due to a notification), then it might end-up in a layer by accident. I think my apprehension are unwarranted though, and we could just do what I suggested without concerning ourselves with strange corner cases, which after all can affect all edit routing.

Reply:
Thank you @pierrebai-adsk,
My question was about the expected behavior for "SessionLayer-" metadata. I was wondering if it's acceptable to allow them to be authored in layers other than the session layer.
As for the implementation, my initial thought was to apply edit routing after the existing `UsdEditContext` that targets the session layer. Would this approach work?:

```c++
    if (isSessionLayerGroupMetadata(_group, &prefixlessGroupName)) {
        PXR_NS::UsdEditContext editCtx(_stage, _stage->GetSessionLayer());
        {
            PXR_NS::TfToken               fullKey(prefixlessGroupName + std::string(":") + _key);
            PrimMetadataEditRouterContext editRouterCtx(prim, PXR_NS::SdfFieldKeys->CustomData, fullKey);
            prim.SetCustomDataByKey(fullKey, ufeValueToVtValue(_value));
        }
    } 
```

Reply:
Re-thinking about it, I like your suggestion as it avoids unnecessary changes to the stage edit target.
I could do that efficiently by adding a `fallbackLayer` argument to `PrimMetadataEditRouterContext`:
```c++
    if (isSessionLayerGroupMetadata(_group, &prefixlessGroupName)) {
        PrimMetadataEditRouterContext ctx(
            prim,
            PXR_NS::SdfFieldKeys->CustomData,
            fullKey,
            /*fallbackLayer=*/_stage->GetSessionLayer());
        prim.SetCustomDataByKey(fullKey, ufeValueToVtValue(_value));
    }
```

Reply:
By default the color data is being written to the SessionLayer since we set the group here: https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/resources/ae/usdschemabase/ae_template.py#L673

But we always intended it to be editRoutable (from your PR). So yes please make it edit routable.

Sean

Reply:
Done.